clients/src/main/java/org/apache/kafka/common/config/AbstractConfig.java

```diff
@@ -342,7 +342,7 @@ public void logUnused() {
      * @param configOverrides Configuration overrides to use.
      * @return The list of configured instances
      */
-    public <T> List<T> getConfiguredInstances(List<String> classNames, Class<T> t, Map<String, Object> configOverrides) {
+    public <T> List<T> getConfiguredInstances(List<?> classNames, Class<T> t, Map<String, Object> configOverrides) {
```

Comment:
We want the class names to be a string, so this change doesn't seem to be desirable.